File: src/main/java/com/example/masterplanbbe/common/response/ErrorResponse.java
package com.example.masterplanbbe.common.response;

import com.example.masterplanbbe.common.exception.ErrorCode;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public class ErrorResponse<T> {
    private final int status;
    private final String message;
    private final T data;

    public static <T> ErrorResponse<T> of(ErrorCode errorCode) {
        return new ErrorResponse<>(errorCode.getStatus(), errorCode.getMessage(), null);
    }

    public static <T> ErrorResponse<T> of(ErrorCode errorCode, T data) {
        return new ErrorResponse<>(errorCode.getStatus(), errorCode.getMessage(), data);
    }
}

File: src/main/kotlin/masterplanb/masterplanbbe/common/response/ErrorResponse.kt
package masterplanb.masterplanbbe.common.response

import masterplanb.masterplanbbe.common.exception.ErrorCode

data class ErrorResponse(
    val status: Int,
    val message: String,
    val code: String
) {
    companion object {
        fun of(errorCode: ErrorCode): ErrorResponse {
            return ErrorResponse(
                status = errorCode.status,
                message = errorCode.message,
                code = errorCode.code
            )
        }
    }
}
